dali/python/nvidia/dali/pipeline.py

```diff
@@ -1078,7 +1086,19 @@ def reset(self):
 
         If pipeline iterator reached the end then reset its state to the beginning.
         """
-        if self._last_iter:
+        if not self._last_iter:
+            # resetting before some external source raised StopIteration is a no-op
+            if self._input_callbacks:
```

Comment:
🤷 I guess it still can happen and we should behave consistently.

Reply:
😟I thought that one would be useful if one has two or more pipelines with (P)ES that raise StopIteration and should be reset but for some reason the epochs diverge. Either because the number of iterations is really (but unintentionally) different in those two or because one uses .run API, with prefetch_queue_depth 1 and resets all pipelines when the first one raises, not letting the others actually reach end of epoch.